server/storage/mvcc/kv_test.go

```diff
@@ -252,6 +253,63 @@ func testKVRangeLimit(t *testing.T, f rangeFunc) {
 		} else if r.Count != int(tt.wcounts) {
 			t.Errorf("#%d: count = %d, want %d", i, r.Count, tt.limit)
 		}
+		if r.More != tt.wmore {
+			t.Errorf("#%d: more = %v, want %v", i, r.More, tt.wmore)
+		}
+	}
+}
+
+func TestKVRangeMaxBytes(t *testing.T)    { testKVRangeMaxBytes(t, normalRangeFunc) }
+func TestKVTxnRangeMaxBytes(t *testing.T) { testKVRangeMaxBytes(t, txnRangeFunc) }
+
+func testKVRangeMaxBytes(t *testing.T, f rangeFunc) {
+	b, tmpPath := betesting.NewDefaultTmpBackend(t)
+	s := NewStore(zaptest.NewLogger(t), b, &lease.FakeLessor{}, StoreConfig{})
+	defer cleanup(s, b, tmpPath)
+
+	kvs := put3TestKVs(s)
+
+	kvsSize := func(kvs []mvccpb.KeyValue) int64 {
+		var s int
+		for _, kv := range kvs {
+			s += kv.Size()
+		}
+		return int64(s)
+	}
+
+	wrev := int64(4)
+	tests := []struct {
+		maxBytes int64
+		wcounts  int64
+		wkvs     []mvccpb.KeyValue
+		wmore    bool
+	}{
+		// no limit
+		{-1, 3, kvs, false},
+		// no limit
+		{0, 3, kvs, false},
+		{kvsSize(kvs[:1]), 3, kvs[:1], true},
+		{kvsSize(kvs[:2]), 3, kvs[:2], true},
+		{kvsSize(kvs), 3, kvs, false},
+		{1000, 3, kvs, false},
+	}
+	for i, tt := range tests {
+		r, err := f(s, []byte("foo"), []byte("foo3"), RangeOptions{MaxBytes: tt.maxBytes})
+		if err != nil {
+			t.Fatalf("#%d: range error (%v)", i, err)
+		}
+		if !reflect.DeepEqual(r.KVs, tt.wkvs) {
+			t.Errorf("#%d: kvs = %+v, want %+v", i, r.KVs, tt.wkvs)
+		}
+		if r.Rev != wrev {
+			t.Errorf("#%d: rev = %d, want %d", i, r.Rev, wrev)
+		}
+		if r.Count != len(kvs) {
+			t.Errorf("#%d: count = %d, want %d", i, r.Count, len(kvs))
+		}
+		if r.More != tt.wmore {
+			t.Errorf("#%d: more = %v, want %v", i, r.More, tt.wmore)
+		}
```

Comment:
Please use golang subtest, there are lots of examples in the repo.  FYI. https://github.com/etcd-io/etcd/pull/14795

Reply:
Done. Thanks!